lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -778,21 +780,52 @@ impl TryFrom<BitmapEvent> for CGOBitmap {
             data_cap: 0,
         };
 
-        // e.decompress consumes e, so we need to call it separately, after populating the fields
-        // above.
-        let mut data = e.decompress()?;
-        res.data_ptr = data.as_mut_ptr();
-        res.data_len = data.len();
-        res.data_cap = data.capacity();
+        let w: u16 = e.width;
+        let h: u16 = e.height;
+
+        let mut encoded = Vec::with_capacity(8192);
+        encode_png(&mut encoded, w, h, e.decompress()?);
+
+        res.data_ptr = encoded.as_mut_ptr();
+        res.data_len = encoded.len();
+        res.data_cap = encoded.capacity();
 
         // Prevent the data field from being freed while Go handles it.
         // It will be dropped once CGOBitmap is dropped (see below).
-        mem::forget(data);
+        mem::forget(encoded);
 
         Ok(res)
```

Comment:
The thing we return here isn't really a `CGOBitmap` so much anymore, it's moreso a `CGOPNG` (the `data_ptr` is no longer in bitmap format, rather it's a png). We should change the name of that structure.

Reply:
Just to be clear- you suggest that we no longer should use CGOBitmap name and just change it to something like CGOPNG?

Reply:
Yes, because it no longer holds a bitmap, it holds a png.